server/routes/archiveGoal/archiveGoalFormValidator.ts

```diff
@@ -0,0 +1,34 @@
+import type { ArchiveGoalForm } from 'forms'
+import formatErrors from '../errorFormatter'
+import ReasonToArchiveGoalValue from '../../enums/ReasonToArchiveGoalValue'
+
+export default function validateArchiveGoalForm(archiveGoalForm: ArchiveGoalForm): Array<Record<string, string>> {
+  const errors: Array<Record<string, string>> = []
+  errors.push(...formatErrors('reason', validateReason(archiveGoalForm)))
+  errors.push(...formatErrors('reasonOther', validateReasonOther(archiveGoalForm)))
+  return errors
+}
+
+const validateReason = (archiveGoalForm: ArchiveGoalForm): Array<string> => {
+  const errors: Array<string> = []
+
+  const { reason } = archiveGoalForm
+  if (!reason) {
+    errors.push(`You must select a reason to archive the goal`)
+  }
+
+  return errors
+}
+
+const validateReasonOther = (archiveGoalForm: ArchiveGoalForm): Array<string> => {
+  const errors: Array<string> = []
+
+  const { reason, reasonOther } = archiveGoalForm
+  if (reason === ReasonToArchiveGoalValue.OTHER && !reasonOther) {
+    errors.push('Enter the reason you are archiving the goal')
+  } else if (reason === ReasonToArchiveGoalValue.OTHER && reasonOther.length > 200) {
+    errors.push(`The reason is limited to 200 characters but you entered ${reasonOther.length}`)
```

Comment:
Where did we get these error messages from? 🤔 
I can't seem them in the ticket (though I didnt look particularly hard, so they might be there)
But I'm particularly interested in the length limit one, in that I dont think we have any other validation messages that say "the limit is x but you entered y" 🤔 
It seems unusual/different terminology to what we do elsewhere I think

Have a look at `stepTitleValidator` as an example, but this is more like our current terminology/pattern:
```
const MAX_STEP_TITLE_LENGTH = 512

export default function validateStepTitle(title?: string): Array<string> {
  const errors: Array<string> = []

  if (!title) {
    errors.push('Enter the step needed to work towards the goal')
  } else if (title.length > MAX_STEP_TITLE_LENGTH) {
    errors.push(`The step description must be ${MAX_STEP_TITLE_LENGTH} characters or less`)
  }

  return errors
}
```

My advice would be to check with the BAs (or dare I say it the designer/content designer, though you might regret that particular avenue of investigation! 🤣 )

Reply:
I've updated error messages and checked with Jo and Karen. I'll be adding the char count element as well after chatting with Karen.